akka-stream/src/main/scala/akka/stream/impl/SplitAfterProcessorImpl.scala

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl
+
+import scala.util.control.NonFatal
+import akka.actor.Props
+import akka.stream.ActorFlowMaterializerSettings
+import akka.stream.Supervision
+import akka.stream.scaladsl.Source
+
+/**
+ * INTERNAL API
+ */
+private[akka] object SplitAfterProcessorImpl {
+  def props(settings: ActorFlowMaterializerSettings, splitPredicate: Any ⇒ Boolean): Props =
+    Props(new SplitAfterProcessorImpl(settings, splitPredicate))
+
+  private trait SplitDecision
```

Comment:
Hmm, what if we add a SplitBefore and SplitAfter, change the function from Any => Boolean to Any => SplitWhere, then we have one unified implementation for splitWhen and splitAfter (which just emit Continue or SplitBefore/SplitWhen depending on the predicate), and also different other splits.